helix-term/src/commands.rs

```diff
@@ -727,6 +728,26 @@ fn goto_file_end(cx: &mut Context) {
     doc.set_selection(view.id, selection);
 }
 
+fn goto_file(cx: &mut Context) {
+    let action = match cx.count() {
+        2 => Action::HorizontalSplit,
+        3 => Action::VerticalSplit,
+        _ => Action::Replace,
+    };
+    let (view, doc) = current_ref!(cx.editor);
+    let text = doc.text();
+    let paths: Vec<_> = doc
+        .selection(view.id)
+        .iter()
+        .map(|r| text.slice(r.from()..r.to()).to_string())
```

Comment:
It would also be useful to auto-detect the filename from the cursor position without explicitly selecting it, like in vim.

Reply:
how do we test there is no selection? Only one empty selection?

Reply:
Hmm, seems that kakoune only uses the current selection: https://github.com/mawww/kakoune/blob/e7100dc87434d933fd8268e5bf70080b627750c5/src/normal.cc#L290-L325

But we can improve on this

Reply:
Added opening current long word as file if no explicit selection based on @sudormrfbin  suggests.